phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -1301,11 +1301,24 @@ public boolean next(List<Cell> results) throws IOException {
         try {
             localScanner = getLocalScanner();
             if (localScanner == null) {
+                region.closeRegionOperation();
                 return false;
             }
             synchronized (localScanner) {
                 if (!shouldVerify()) {
                     skipped = true;
+                    region.closeRegionOperation();
```

Comment:
@swaroopak - do we need the region.closeRegionOperation() and localScanner.close() calls? They appear to already be called in the finally block, which will execute even if we return false early. And a finally block seems like a safer place for them so some other code path doesn't miss them in the future.

Reply:
As I stated before, the fix is to check the isRegionClosingOrSplitting attribute of the UngroupedAggregateRegionObserver coproc and return an IOException after every scanner next or table batch operation done during index rebuild or verification if  the region is closing or splitting.